### PR TITLE
docs(TechDocs): Update docs around AWS SDK downgrade to v2

### PR DESCRIPTION
**Pull request by OrkoHunter**
Follow up of https://github.com/backstage/backstage/pull/4434

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4552-ithbbswbdsuea9l48u8nrwni0bmoin5p.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602c091ffcb22d582756f5f8

File: docs/features/techdocs/using-cloud-storage.md [hunk 1]
```diff
@@ -142,22 +142,30 @@ variables**
...
 
 - `AWS_ACCESS_KEY_ID`
 - `AWS_SECRET_ACCESS_KEY`
 - `AWS_REGION`
 
 are set and can be used to access the bucket you created in step 2, they will be
-used by the AWS SDK v3 Node.js client for authentication.
-[Refer to the official documentation.](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-environment.html)
+used by the AWS SDK V2 Node.js client for authentication.
+[Refer to the official documentation.](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-environment.html)
```
Comment (realandersn) [inline docs/features/techdocs/using-cloud-storage.md:155]: Perhaps provide better context in link. Example: Refer to the [official AWS documentation for loading credentials in Node.js from environment variables](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-environment.html) (or similar).

File: docs/features/techdocs/using-cloud-storage.md [hunk 2]
```diff
@@ -142,22 +142,30 @@ variables**
...
 If the environment variables are missing, the AWS SDK tries to read the
 `~/.aws/credentials` file for credentials.
-[Refer to the official documentation.](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/loading-node-credentials-shared.html)
+[Refer to the official documentation.](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-shared.html)
+
+If you are using Amazon EC2 instance to deploy Backstage, you do not need to
+obtain the access keys separately. They can be made available in the environment
+automatically by defining appropriate IAM role with access to the bucket. Read
+more
+[here](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html#use-roles).
```
Comment (realandersn) [inline docs/features/techdocs/using-cloud-storage.md:165]: Same: maybe For more information, see [official AWS documentation for using IAM roles](https://docs.aws.amazon.com/general/latest/gr/aws-access-keys-best-practices.html#use-roles).

**Review by realandersn — COMMENTED** [review]
Small comments on links.

File: docs/features/techdocs/using-cloud-storage.md (hunk 1 above)
Comment (OrkoHunter) [reply docs/features/techdocs/using-cloud-storage.md]: Thanks, yes!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4552-ithbbswbdsuea9l48u8nrwni0bmoin5p.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602c091ffcb22d582756f5f8

**Review by realandersn — APPROVED** [review]

**Pull request merged**